### PR TITLE
Add new CFG Array version (5.5)

### PR DESCRIPTION
**Pull request by klarich**
Changes include: 

- Created a new function called clean_glycan_string to assist with removing small errors from CFG glycan strings
- Added an example script for generating a new glycan array file
- Updated README to describe the process for adding a new glycan array file 
- Added new glycan array version csv's to `Data/CFG_Array_Versions` and `ccarl/glycan_parsers/CFG_Array_Versions` - note: I'm not sure if it is necessary to add the csv to both locations or if it only needs to be added to one of those locations, however, since all the csv files were in both locations I added it to both. 

I have not built the Dockerfile or tested the code yet as I was running into some errors building the image on my machine - but I can dig into this a bit more.

### CODE REVIEW

File: Scripts/generate_new_glycan_array_version.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    print(f"Failed to parse {len(failed)} structures. {len(succeeded)} parsed successfully ")
+    print(f"Failed structures that will need to be fixed manually:")
+    for structure in failed:
+        print(structure)
+
+    if not failed:  # Only write new glycan array file if all structures were successfully parsed.
+        with open(output_path, 'w') as f:
+            for chart_number, structure in zip(df['Chart Number'], succeeded):
+                f.write(f"{chart_number},{structure}\n")
```
Comment (andrewguy) [inline Scripts/generate_new_glycan_array_version.py:36]:
```suggestion
        df['Structure'] = succeeded
        df.to_csv(output_path, columns=["Chart Number", "Structure"], index=False, header=False)
```

Using pandas to write to csv will handle any issues with escaping `,` characters.

File: ccarl/glycan_parsers/cfg_array_versions.py [hunk 2]
```diff
@@ -55,3 +53,17 @@ def get_likely_cfg_array_version(glycan_list, distance_threshold=2.0):
     if scaled_levenshtein_total > distance_threshold:
         raise ValueError("Glycan list does not match to known array versions.")
     return list(cfg_array_versions[likely_array][1]), likely_array, likely_array_mismatches, scaled_levenshtein_total
+
+
+def clean_glycan_string(glycan: str) -> str:
+    """
+    Method to clean glycan strings before parsing. n
```
Comment (andrewguy) [inline ccarl/glycan_parsers/cfg_array_versions.py:60]:
```suggestion
    Method to clean glycan strings before parsing.
    
```

File: Data/CFG_Array_Versions/Array_5.5.csv [hunk 3]
```diff
@@ -0,0 +1,562 @@
...
+39,(6S)(4S)Galb1-4GlcNAcb-Sp0
+40,(4S)Galb1-4GlcNAcb-Sp8
+41,(6P)Mana-Sp8
+42,(6S)Galb1-4Glcb-Sp0
+43,(6S)Galb1-4Glcb-Sp8
+44,(6S)Galb1-4GlcNAcb-Sp8
+45,(6S)Galb1-4(6S)Glcb-Sp8
+46,Neu5Aca2-3(6S)Galb1-4GlcNAcb-Sp8
+47,(6S)GlcNAcb-Sp8
+48,Neu5,9Ac2a-Sp8
```
Comment (andrewguy) [inline Data/CFG_Array_Versions/Array_5.5.csv:48]:
```suggestion
48,"Neu5,9Ac2a-Sp8"
```

File: Data/CFG_Array_Versions/Array_5.5.csv [hunk 4]
```diff
@@ -0,0 +1,562 @@
...
+49,Neu5,9Ac2a2-6Galb1-4GlcNAcb-Sp8
```
Comment (andrewguy) [inline Data/CFG_Array_Versions/Array_5.5.csv:49]:
```suggestion
49,"Neu5,9Ac2a2-6Galb1-4GlcNAcb-Sp8"
```

File: Data/CFG_Array_Versions/Array_5.5.csv [hunk 5]
```diff
@@ -0,0 +1,562 @@
...
+309,Neu5Aca2-3Galb1-4GlcNAcb1-6(Neu5Aca2-3Galb1-3)GalNAca-Sp14
+310,Neu5Aca2-6Galb1-4GlcNAcb1-2Mana1-6(Neu5Aca2-3Galb1-4GlcNAcb1-2Mana1-3)Manb1-4GlcNAcb1-4GlcNAcb-Sp12
+311,Galb1-4GlcNAcb1-2Mana1-6(Neu5Aca2-6Galb1-4GlcNAcb1-2Mana1-3)Manb1-4GlcNAcb1-4GlcNAcb-Sp12
+312,Neu5Aca2-8Neu5Aca2-8Neu5Acb-Sp8
+313,Neu5Gcb2-6Galb1-4GlcNAc-Sp8
+314,Galb1-3GlcNAcb1-2Mana1-6(Galb1-3GlcNAcb1-2Mana1-3)Manb1-4GlcNAcb1-4GlcNAcb-Sp19
+315,Neu5Aca2-3Galb1-4GlcNAcb1-2Mana1-6(Neu5Aca2-3Galb1-4GlcNAcb1-2Mana1-3)Manb1-4GlcNAcb1-4GlcNAcb-Sp12
+316,Neu5Aca2-3Galb1-4GlcNAcb1-2Mana1-6(Neu5Aca2-6Galb1-4GlcNAcb1-2Mana1-3)Manb1-4GlcNAcb1-4GlcNAcb-Sp12
+317,Galb1-4(Fuca1-3)GlcNAcb1-2Mana1-6(Galb1-4(Fuca1-3)GlcNAcb1-2Mana1-3)Manb1-4GlcNAcb1-4GlcNAcb-Sp20
+318,Neu5,9Ac2a2-3Galb1-3GlcNAcb-Sp0
```
Comment (andrewguy) [inline Data/CFG_Array_Versions/Array_5.5.csv:318]:
```suggestion
318,"Neu5,9Ac2a2-3Galb1-3GlcNAcb-Sp0"
```

File: Scripts/generate_new_glycan_array_version.py (hunk 1 above)
Comment (klarich) [reply Scripts/generate_new_glycan_array_version.py]: Ah good catch

**Review by andrewguy — APPROVED** [review]

**Pull request merged**